### ISSUE TITLE
Unable to debug on MacOS even after signing r2

### ISSUE THREAD

**Issue by vinofou** [radareorg/radare2#18258]
## Environment

```sh
Mer 20 jan 2021 22:41:45 EST
radare2 5.0.1-git 25607 @ darwin-x86-64 git.5.0.0
commit: d8b6f52e1ddccf3a6051335e399fb150f23065a8 build: 2021-01-20__22:28:34
Darwin x86_64
```

## Description

Hi, I keep getting these errors while trying to do `r2 -d app1` :

![image](https://user-images.githubusercontent.com/44421022/105277676-dcdf8400-5b71-11eb-9070-baf5ed27415a.png)

Even if it says that r2 may possibly be unsigned sometimes, I can assure that I followed all the steps to manually sign it from [this link](https://github.com/radareorg/radare2/blob/master/doc/macos.md).

## Test

There isn't really any steps that can be done from your side but I will provide more informations when needed. I am starting in the Assembly world so I may be missing something here.

Thank you.

**Comment by trufae** [comment]
How did you build radare?

**Comment by vinofou** [comment]
I actually tried both homebrew `brew install radare2` and manually with cloning the repository then `/sys/install.sh`.

**Comment by vinofou** [comment]
Here's the output of the `rabin2 -I` command so you can verify if there's an incompatibility : 
`arch     x86
baddr    0x8048000
binsz    6421
bintype  elf
bits     32
canary   true
class    ELF32
compiler GCC: (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609
crypto   false
endian   little
havecode true
intrp    /lib/ld-linux.so.2
laddr    0x0
lang     c
linenum  true
lsyms    true
machine  Intel 80386
maxopsz  16
minopsz  1
nx       true
os       linux
pcalign  0
pic      false
relocs   true
relro    partial
rpath    NONE
sanitiz  false
static   false
stripped false
subsys   linux
va       true`

Do you see anything wrong with this ?

**Comment by trufae** [comment]
The original issue complains about osx not being able to debug and your last message shows an x86-32 ubuntu binary. Im confused. You should use the codesign tool with -dvvv argument to see the signing info of the executable. Try running make purge before sys/install.sh that may be deleting the brew installation. Also are you sure you r running r2 you built and not the one bew built? Just check the PATH and see whats up.

read doc/macos.md and create the developer certificate if you dont have any and use this to sign the binaries. Sys/install.sh will do yhat if you follow the steps and give the right name to it.

also. If you have sip enabled you wont be able to debug any executable outside your home. So r2 -d r2 is probably the easiest way to test that

**Comment by vinofou** [comment]
This compiled file was originally given to me by a teacher with a 32-bit Ubuntu VM so I could practice debugging. I thought that by copying it on my OSX machine, I could be able to debug it right away. Does it need to be recompiled on MacOS ? I tried to take a look online but it seems like it is not possible to compile a 32-bit file on the latest version of MacOS because of the Xcode command line tools version. I also have the source code if that can help. 

I am pretty sure I am using the right version of radare2 has the homebrew version has been completely uninstalled and cannot be found on my system when I list programs. 

I ran the codesign tool as asked and it tells me that the file is not signed at all. Is it because I did something wrong will following the manual steps on the doc/macos.md ? I believe I am missing a step here as I did not sign my app1 file, I only followed the manual steps at the middle of the page. 

When you say that I need to disable sip, I understood that it was only for apple-signed binaries. I thought it meant that is wasn't possible for files from Apple but it was okay for compiled code from the user. If I am in subfolders in the home directory as shown on my screenshots, is that okay or I need to exactly in the home folder ?

Sorry if I am asking a lot of questions at the same time. I am genuinely interested in getting it working, but I am starting from the beginning so that's why I may be missing a couple things here. Thank you for taking the time helping me.
